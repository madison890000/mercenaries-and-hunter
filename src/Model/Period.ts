import Company from './Company';
import DataModel from './types';
import Base from './Base';

interface IPeriod {
    start: Date;
    end?: Date;
    keywords: DataModel.Keyword[];
    company: Company;
    jobPositionLevel: DataModel.JobPositionLevel;
    jobPosition: DataModel.JobPosition;
    achievements: DataModel.Achievement[];
    jobSummaries: string[];
    solutionsOfHowToImplement: DataModel.SolutionsOfHowToImplement[];
}

export default class Period extends Base {
    public start: Date;
    public end?: Date;
    public keywords: DataModel.Keyword[];
    public achievements: DataModel.Achievement[];
    public jobSummaries: string[];
    public company: Company;
    public jobPositionLevel: DataModel.JobPositionLevel;
    public jobPosition: DataModel.JobPosition;
    public solutionsOfHowToImplement: DataModel.SolutionsOfHowToImplement[];

    constructor({ solutionsOfHowToImplement, company, jobSummaries, jobPosition, jobPositionLevel, achievements, keywords, start, end }: IPeriod) {
        super();
        this.company = company;
        this.start = start;
        this.end = end;
        this.achievements = achievements;
        this.jobSummaries = jobSummaries;
        this.keywords = keywords;
        this.jobPositionLevel = jobPositionLevel;
        this.jobPosition = jobPosition;
        this.solutionsOfHowToImplement = solutionsOfHowToImplement;
    }
}
