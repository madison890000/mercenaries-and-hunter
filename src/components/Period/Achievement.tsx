import Tag from '../Tag';
import React from 'react';

interface AchievementProps {
    title: string;
    categories: string[];
}

const Achievement = ({ title, categories }: AchievementProps) => (
    <li>
        {title}
        <span>
            {categories?.map(c => (
                <Tag>{c}</Tag>
            ))}
        </span>
    </li>
);

export default Achievement;
