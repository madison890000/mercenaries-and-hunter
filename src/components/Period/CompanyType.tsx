import Tag from '../Tag';
import React from 'react';
import DataModel from '../../models/types';

interface CompanyTypeProps {
    companyType: DataModel.CompanyType;
}

const CompanyType = ({ companyType }: CompanyTypeProps) => (
    <span>
        {companyType === DataModel.CompanyType.Enterprise && <Tag type="less">Enterprise</Tag>}
        {companyType === DataModel.CompanyType.Startup && <Tag type="less">Startup</Tag>}
    </span>
);

export default CompanyType;
