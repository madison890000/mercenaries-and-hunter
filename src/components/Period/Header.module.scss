.header {
    font-weight: bold;

    & > span {
        margin-right: var(--inline-block-margin);
    }

    @media screen and (max-width: 414px) {
        width: 300px;

        & > span {
            display: block;
        }
    }
}

.job {
    & > span {
        margin-right: var(--inline-block-margin);
    }
}

.companyName {
    font-weight: normal;
    color: var(--period-sub-title-color);
}
