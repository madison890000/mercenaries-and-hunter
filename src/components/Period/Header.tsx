import Tag from '../Tag';
import React from 'react';
import DataModel from '../../models/types';
import styles from './Header.module.scss';
import StringWithID from '../../models/StringWithID';
import capitalize from '../../utils/capitalize';

interface HeaderProps {
    keywords: StringWithID[];
    jobPositionLevel: DataModel.JobPositionLevel;
    jobPosition: DataModel.JobPosition;
    companyName: string;
    companyType: DataModel.CompanyType;
}

const Header = ({ keywords, companyName, jobPosition, jobPositionLevel }: HeaderProps) => (
    <div className={styles.header}>
        <span className={styles.job}>
            {jobPositionLevel && <span>{capitalize(jobPositionLevel.toLowerCase())}</span>}
            <span>{jobPosition}</span>
        </span>
        <span>
            {keywords?.map(keyword => (
                <Tag type="filled" key={keyword.id}>
                    {keyword.toString()}
                </Tag>
            ))}
        </span>
        <span className={styles.companyName}>
            <span>{companyName}</span>
        </span>
    </div>
);

export default Header;
