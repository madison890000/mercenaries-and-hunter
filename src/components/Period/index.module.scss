.period {
  display: flex;
  align-items: flex-start;
  justify-content: space-between;

  @media screen and (max-width: 414px) {
    flex-direction: column-reverse;
  }
}

.timeline {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  @media screen and (max-width: 414px) {
    width: 100%;
    justify-content: space-between;
    flex-direction: row-reverse;
  }
}

.header {
  font-weight: bold;

  & > span {
    margin-right: var(--inline-block-margin);
  }

  @media screen and (max-width: 414px) {
    width: 300px;

    & > span {
      display: block;
    }
  }
}

.job {
  & > span {
    margin-right: var(--inline-block-margin);
  }
}

.companyName {
  font-weight: normal;
  color: gray;
}

.achievements {
  //border: 1px solid gray;
  font-weight: bold;
}
