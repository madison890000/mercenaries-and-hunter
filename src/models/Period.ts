import Company from './Company';
import DataModel from './types';
import Base from './Base';
import StringWithID from './StringWithID';
import { v4 } from 'uuid';

interface IPeriod {
    start: Date;
    end?: Date;
    keywords: string[];
    company: Company;
    jobPositionLevel: DataModel.JobPositionLevel;
    jobPosition: DataModel.JobPosition;
    achievements: DataModel.IAchievement[];
    jobSummaries: string[];
    solutionsOfHowToImplement: DataModel.SolutionsOfHowToImplement[];
}

export default class Period extends Base {
    public start: Date;
    public end?: Date;
    public keywords: StringWithID[];
    public achievements: DataModel.Achievement[];
    public jobSummaries: StringWithID[];
    public company: Company;
    /**
     * @deprecated please use Job.level
     */
    public jobPositionLevel: DataModel.JobPositionLevel;
    /**
     * @deprecated please use Job.position
     */
    public jobPosition: DataModel.JobPosition;
    public job: DataModel.Job;
    public solutionsOfHowToImplement: DataModel.SolutionsOfHowToImplement[];

    constructor({
        solutionsOfHowToImplement,
        company,
        jobSummaries,
        jobPosition,
        jobPositionLevel,
        achievements,
        keywords,
        start,
        end
    }: IPeriod) {
        super();
        this.company = company;
        this.start = start;
        this.end = end;
        this.achievements = achievements.map(achievement => ({
            id: v4(),
            text: achievement.text,
            categories: achievement.categories.map(e => new StringWithID(e))
        }));
        this.jobSummaries = jobSummaries.map(e => new StringWithID(e));
        this.keywords = keywords.map(e => new StringWithID(e));
        this.jobPositionLevel = jobPositionLevel;
        this.jobPosition = jobPosition;
        this.job = {
            position: jobPosition,
            level: jobPositionLevel
        };
        this.solutionsOfHowToImplement = solutionsOfHowToImplement;
    }
}
