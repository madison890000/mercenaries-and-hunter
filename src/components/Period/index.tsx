import React from 'react';
import styles from './index.module.scss';
import DataModel from '../../models/types';
import Times from '../Times';
import Achievement from './Achievement';
import TimeLineItem from '../TimeLines/TimeLineItem';
import { defineMessages, useIntl } from 'react-intl';
import Header from './Header';
import StringWithID from '../../models/StringWithID';
import capitalize from '../../utils/capitalize';

interface PeriodProps {
    start: Date;
    periodColor: string;
    end?: Date;
    keywords: StringWithID[];
    companyName: string;
    companyType: DataModel.CompanyType;
    jobPositionLevel: DataModel.JobPositionLevel;
    jobPosition: DataModel.JobPosition;
    achievements: DataModel.Achievement[];
    jobSummaries: StringWithID[];
}

const messages = defineMessages({
    jobSummaries: {
        id: 'component.period.jobSummaries',
        defaultMessage: 'Job summaries'
    },
    achievements: {
        id: 'component.period.achievements',
        defaultMessage: 'Achievements'
    }
});
const Period = ({
    start,
    periodColor,
    end,
    companyName,
    companyType,
    keywords,
    jobPositionLevel,
    jobPosition,
    achievements,
    jobSummaries
}: PeriodProps) => {
    const intl = useIntl();
    return (
        <div className={styles.periodContainer}>
            <div className={styles.period}>
                <div>
                    <Header
                        jobPosition={jobPosition}
                        companyName={companyName}
                        companyType={companyType}
                        keywords={keywords}
                        jobPositionLevel={jobPositionLevel}
                    />
                </div>
                <div className={styles.timeline}>
                    <TimeLineItem start={start} end={end} periodColor={periodColor} />
                    <Times start={start} end={end} />
                </div>
            </div>

            <div>
                <h5>{intl.formatMessage(messages.jobSummaries)}:</h5>
                <ul>
                    {jobSummaries?.map(summary => (
                        <li key={summary.id}>{capitalize(summary.toString())}</li>
                    ))}
                </ul>
            </div>
            <div>
                <h5>{intl.formatMessage(messages.achievements)}:</h5>
                <ul className={styles.achievements}>
                    {achievements?.map(achievement => (
                        <Achievement
                            key={achievement.id}
                            title={achievement?.text}
                            categories={achievement?.categories}
                        />
                    ))}
                </ul>
            </div>
        </div>
    );
};

export default Period;
