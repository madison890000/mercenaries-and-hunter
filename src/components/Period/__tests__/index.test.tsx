import React from 'react';
import Period from '../index';
import renderer from 'react-test-renderer';
import DataModel from '../../../Model/types';

test('renders Period', () => {
    const testCompanyName = 'test company name';
    const start = new Date('2012-01-01');
    const end = new Date('2013-01-01');
    const keywords = ['React'] as DataModel.Keyword[];
    const jobPositionLevel = DataModel.JobPositionLevel.Senior;
    const jobPosition = 'Software Engineer';
    const achievements = [] as DataModel.Achievement[];
    const jobSummaries = [] as string[];
    const PeriodDom = renderer
        .create(
            <Period
                start={start}
                end={end}
                keywords={keywords}
                jobPositionLevel={jobPositionLevel}
                jobPosition={jobPosition}
                achievements={achievements}
                jobSummaries={jobSummaries}
                companyName={testCompanyName}
                periodColor={'red'}
            />
        )
        .toJSON();
    expect(PeriodDom).toMatchSnapshot();
});
