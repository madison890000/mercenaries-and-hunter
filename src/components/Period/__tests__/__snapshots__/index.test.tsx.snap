// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Period 1`] = `
Array [
  <div
    className="period"
  >
    <div>
      <div
        className="header"
      >
        <span
          className="job"
        >
          <span>
            Senior
          </span>
          <span>
            Software Engineer
          </span>
        </span>
        <span
          className="companyName"
        >
          <span>
            test company name
            <span />
          </span>
        </span>
        <span>
          <span
            className="filled"
          >
            React
          </span>
        </span>
      </div>
    </div>
    <div
      className="timeline"
    >
      <svg
        height="20"
        style={
          Object {
            "backgroundColor": "",
          }
        }
        width={78}
      >
        <g>
          <rect
            height="10"
            width={5}
            x={0}
            y="5"
          />
          <rect
            height="10"
            width={5}
            x={6}
            y="5"
          />
          <rect
            height="10"
            width={5}
            x={12}
            y="5"
          />
          <rect
            height="10"
            width={5}
            x={18}
            y="5"
          />
          <rect
            height="10"
            width={5}
            x={24}
            y="5"
          />
          <rect
            height="10"
            width={5}
            x={30}
            y="5"
          />
          <rect
            height="10"
            width={5}
            x={36}
            y="5"
          />
          <rect
            height="10"
            width={5}
            x={42}
            y="5"
          />
          <rect
            height="10"
            width={5}
            x={48}
            y="5"
          />
          <rect
            height="10"
            width={5}
            x={54}
            y="5"
          />
          <rect
            height="10"
            width={5}
            x={60}
            y="5"
          />
          <rect
            height="10"
            width={5}
            x={66}
            y="5"
          />
          <rect
            height="10"
            width={5}
            x={72}
            y="5"
          />
        </g>
      </svg>
      <div
        className="times"
      >
        <span>
          Jan, 2012
        </span>
        <span>
          -
        </span>
        <span>
          Jan, 2013
        </span>
      </div>
    </div>
  </div>,
  <div>
    <h5>
      Job summaries:
    </h5>
    <ul />
  </div>,
  <div>
    <h5>
      Achievements:
    </h5>
    <ul
      className="achievements"
    />
  </div>,
]
`;
